### ISSUE TITLE
HAML parser doesn't include full comment text in block comments

### ISSUE THREAD

**Issue by Ajedi32** [haml/haml#838]
The HAML parser currently provides an interface for accessing the text contents of HAML comments:

```ruby
engine = Haml::Engine.new <<INPUT
 -# This is a comment!
%p This is a paragraph.
INPUT
document_root = engine.parser.root
document_root.children.first.type #=> :haml_comment
document_root.children.first.value #=> {:text=>" This is a comment!"}
```

Unfortunately, this doesn't work correctly for block-level comments:

```ruby
engine = Haml::Engine.new <<INPUT
-#
  This is a comment block
  Text in this block is ignored.
%p This is a paragraph.
INPUT
document_root = engine.parser.root
document_root.children.first.type #=> :haml_comment
document_root.children.first.value #=> {:text=>""}
```

Oh, and for those who might ask why I need to access the text of comments, since comments typically should just be ignored: I'm trying to build a system similar to Front Matter in Jekyll which respects HAML's syntax by putting Front Matter in HAML comments and not encouraging people to put non-HAML syntax in HAML files.

**Comment by mattwildig** [comment]
I’ve fixed this in b09e4589ad0b9f21779854cf0eb7ceb04e171964, but the fix is really a side effect of the work there. This is really an internal detail, I don’t think we can guarantee that the interface won’t change in the future. (Having said that, it’s been fairly stable for a while and there are no current plans to change it as far as I know, so you’ll probably be okay for now).

**Comment by Ajedi32** [comment]
Okay, thanks for the warning.
